File: swift/problem941.swift
class Solution {
    func validMountainArray(_ A: [Int]) -> Bool {
        let length = A.count
        
        if (length < 3 || A[1] < A[0]) {
            return false
        }
        
        var flag = true; var prev = A[1]
        
        for i in 2..<length {
            if flag {
                if (A[i] < prev) {
                    flag = false
                } else if (A[i] == prev){
                    return false
                }
            } else {
                if (A[i] >= prev) {
                    return false
                }
            }
            prev = A[i]
        }
        
        return !flag
    }
}

File: python/problem941.py
class Solution:
    def validMountainArray(self, A: List[int]) -> bool:
        k = 0
        if not A or len(A) == 1:
            return False
        if A[0] > A[1]:
            return False
        for i in range(1,len(A)-1):
            if A[i] > A[i+1]:
                k = i+1
                break
                
        for i in range(k,len(A)-1):
            if A[i] <= A[i+1]:
                return False
            
        return True
        